src/ess/dream/io/geant4.py

```diff
@@ -262,13 +262,16 @@ def assemble_detector_data(
     da.bins.coords["tof"] = da.bins.coords["tof"].to(unit="us")
 
     period = (1.0 / sc.scalar(14.0, unit="Hz")).to(unit="us")
-    # Bin the data into bins with a 71ms period
-    da = da.bin(tof=sc.arange("tof", 3) * period)
+    # Bin the data into bins with a 71ms period.
+    npulses = int((da.bins.coords["tof"].max() / period).value)
+    da = da.bin(tof=sc.arange("tof", npulses + 1) * period)
```

Comment:
```suggestion
    npulses = int((da.bins.coords["tof"].max() / period).ceil().value)
    da = da.bin(tof=sc.arange("tof", npulses) * period)
```
So that `npulses` actually is the number of pulses. Without this, I think `[:npulses]` below is incorrect.